src/server.cpp

```diff
@@ -431,6 +431,18 @@ CServer::CServer ( const int          iNewMaxNumChan,
         SIGNAL ( CLReqConnClientsList ( CHostAddress ) ),
         this, SLOT ( OnCLReqConnClientsList ( CHostAddress ) ) );
 
+    QObject::connect ( &ConnLessProtocol,
+        SIGNAL ( CLReqChannelLevelList ( CHostAddress, bool ) ),
+        this, SLOT ( OnCLReqChannelLevelList ( CHostAddress, bool ) ) );
+
+    qRegisterMetaType< Int16_t >();
+    qRegisterMetaType< CVector< int16_t > >();
+    qRegisterMetaType< CVector< CVector< int16_t > > >();
+    QObject::connect ( this,
```

Comment:
Hmm. You are including the entire audio buffers in the message. Do you think we can get performance issues by this?

Reply:
I guess I could pass the buffer to the calculation method and then emit just the result.  That would save the copy but run the calculation in the main thread (although, only once every 200 frames).

Reply:
It would also save having to define `Int16_t`, too, as that would be passed directly to the method.

Reply:
OK, I've moved where the calculation is done.  I don't think (given it's only for opted in clients and only if at least one is opted in) it'll cause too much CPU stress and it does save the CPU and memory stress of the large copy operation, so I'd agree it's a win.

Reply:
Great, thanks.